updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,167 @@
++
+[source,terminal]
+----
+$ oc label node <node name> node-role.kubernetes.io/<mcp name>=
+----
++
+[source,terminal]
+----
+$ oc label node <node name>  node-role.kubernetes.io/worker-
+----
+. Create a the machineconfig pool
++
+[source,yaml]
+----
+$ cat mcp_foo.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: mcp_foo | <1>
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,mcp_foo]} | <1>
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/mcp_foo: "" | <1>
+----
+<1> MCP name
++
+[source,terminal]
+----
+$ oc create -f mcp_foo.yaml
+----
++
+. To see the list of MCPs present in the cluster and their state
++
+[source,terminal]
+----
+$ oc get mcp
+----
+
+=== Pause a MCP
+
+Pausing a MCP will keep it from updating to the new OS version by the machine config operator.
+
+[source,terminal]
+----
+$ oc patch mcp/mcp_foo --patch '{"spec":{"paused":true}}' --type=merge
+----
+
+=== Unpause a MCP
+
+Unpausing will enable the nodes in the MCP to move to the new OS version and reboot (if required) it.
+
+[source,terminal]
+----
+$ oc patch mcp/mcp_foo --patch '{"spec":{"paused":false}}' --type=merge
+----
+
+== Steps to remove a node from a MCP
+
+A node must have a role to be properly functioning with in the OpenShift cluster.
+If you want to remove a node from a MCP, you should first relabel the node as a worker as it should be part of worker MCP if is not going to be part of any other MCP. Once it is labeled as worker only then proceed to unlabel it from the MCP.
+
+. Label it as worker
++
+[source,terminal]
+----
+$ oc label node <node name>  node-role.kubernetes.io/worker=
+----
++
+. Remove the MCP label
++
+[source,terminal]
+----
+$oc label node <node name> node-role.kubernetes.io/<mcp name>-
+----
++
+. The machine config operator is then going to reconcile the node to the worker pool configuration. Check the output of `oc get mcp` to make sure the worker pool is updated before going to the next step.
++
+[source,terminal]
+----
+$ oc get mcp
+NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED
+mcp_foo  rendered-mcp_foo-e13e08651cde58bdfcaeb2f6f2a5b23c  True      False      False
+master   rendered-master-63f780fa30a02f897ffcd2d6c3fba467   True      False      False
+worker   rendered-worker-e13e08651cde58bdfcaeb2f6f2a5b23c   False     True       False
+----
++
+. Delete the MCP
++
+[source,terminal]
+----
+$ oc delete mcp mcp_foo
+----
+
+== In Case Of Failure
+
+In case of failure, keep all the MCP paused and wait for the version with the bug fix and start the update process again.
+
+[NOTE]
+====
+We do not recommend updating MCPs to different versions i.e. one MCP from 4.Y.100 to 4.Y+1.10 and another 4.Y.100 to 4.Y+1.20.
+This scenario is never tested and may result in to undefined cluster state.
```

Comment:
I don't see any obvious problems with this sort of split.  But one issue would be that until you put the canary pool(s) on 4.(y+1).20, you are probably not going to want to put your production pools on that release (for folks who are conservative and non-Kube-native enough to have canary pools in the first place).  So maybe gentler reminder "hey, if you have a canary pool, don't forget to use it to test the nominal fix too" instead of your current warning about... effectively swapping one canary pool for another... being not recommended?

Reply:
Unless we are testing this scenario we can not say this situation will be ok. Also MCO folks and architects mentioned this specifically. There were concern that it will lead to situations which can not be predicted.

Reply:
In the past we've said as long as all sets of nodes are moving along valid upgrade edges we're OK.

So as long as there are edges from 4.Y.100 to 4.Y+1.10 AND from 4.Y.100 to 4.Y+1.20 we're fine but this does get really complicated to reason about. They should avoid situations where they never converge because they're going to have to evaluate edges across all of these sets until they do. So ideally they'd allow that first MCP to upgrade from 4.Y+1.10 to 4.Y+1.20 before doing anyhting else.

This stalling is part of the reason we've made sure that z-streams are always cumulative, it eases this considerably.

I'm not sure if we should keep this note or just simplify it to say that all MCPs need to upgrade along edges that exist in the graph without going into much more detail.

Reply:
> 'm not sure if we should keep this note or just simplify it to say that all MCPs need to upgrade along edges that exist in the graph without going into much more detail.

Yeah, it is confusing to understand unless some one has very clear understanding of how the graphs are decided.

Reply:
@LalatenduMohanty @sdodson What is the decision here? Leave the note or remove it?

Reply:
We will go ahead with the current note. We will probably improve the text in future.